client/components/date-picker/day.jsx

```diff
@@ -1,87 +1,115 @@
 /**
  * External Dependencies
  */
-var React = require( 'react' ),
-	noop = require( 'lodash/noop' ),
-	classNames = require( 'classnames' );
+import React, { PropTypes, Component } from 'react';
+import { noop, map } from 'lodash';
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var Tooltip = require( 'components/tooltip' );
-
-module.exports = React.createClass( {
-	displayName: 'DatePickerDay',
-
-	propTypes: {
-		date: React.PropTypes.object.isRequired,
-		events: React.PropTypes.array
-	},
-
-	getInitialState: function() {
-		return {
-			showTooltip: false
-		};
-	},
-
-	isPastDay: function( date ) {
-		var today = this.moment().set( {
+import Tooltip from 'components/tooltip';
+import { CalendarEvent } from './event';
+
+class DatePickerDay extends Component {
+	static propTypes = {
+		date: PropTypes.object.isRequired,
+		events: PropTypes.array,
+		maxEventsPerTooltip: PropTypes.number,
+	};
+
+	static defaultProps = {
+		maxEventsPerTooltip: 8,
+	};
+
+	state = {
+		showTooltip: false,
+	};
+
+	isPastDay( date ) {
+		const today = this.props.moment().set( {
 			hour: 0,
 			minute: 0,
 			second: 0,
 			millisecond: 0
 		} );
 
 		date = date || this.props.date;
-
 		return ( +today - 1 ) >= +date;
-	},
+	}
+
+	showTooltip = () => {
+		if ( ! this.props.events.length ) {
+			return;
+		}
 
-	handleTooltip: function( show ) {
-		var showTooltip = ! ! this.props.events.length && show;
-		this.setState( { showTooltip: showTooltip } );
-	},
+		this.setState( { showTooltip: true } );
+	}
 
-	renderTooltip: function() {
-		var label;
+	hideTooltip = () => {
+		this.setState( { showTooltip: false } );
+	}
 
+	renderTooltip() {
 		if ( ! this.state.showTooltip ) {
 			return;
 		}
 
-		label = this.translate(
-				'%(posts)d post',
-				'%(posts)d posts', {
-					count: this.props.events.length,
-					args: {
-						posts: this.props.events.length
-					}
+		const label = this.props.translate(
+			'%(events)d event',
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( '%d Event', '%d Events' )` *ES Score: 32*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).